docs/concepts/glossary.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Glossary
+nav-pos: 3
+nav-title: Glossary
+nav-parent_id: concepts
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+#### Flink Application Cluster
+
+A Flink Application Cluster is a dedicated [Flink Cluster](./glossary#flink-cluster) that only
+executes a single [Flink Job](./glossary#flink-job). The lifetime of the
+[Flink Cluster](./glossary#flink-cluster) is bound to the lifetime of the Flink Job. Formerly
+Flink Application Clusters were also known as Flink Clusters in *job mode*.
+
+#### Flink Cluster
+
+The distributed system consisting of (typically) one Flink Master process and one or more Flink
+Taskmanagers processes.
```

Comment:
In my opinion, we should use "Flink Master (process)" instead of Flink JobManager, yes. Flink JobManager is too overloaded, since the FLIP-6 effort. Are there cases, where the JobManagers are not running in the same process as Resource Manager and Dispatcher?

I also thought about adding these to the Glossary. We should also add JobManager then, to avoid confusion, so: Flink Master, JobManager, TaskManager. 

What do you think?

Reply:
In theory, yes: JobManager, ResourceManager and Dispatcher can be deployed separately...Flink's own startup scripts are not (yet?) making use of this though, afaik.

Unless you want to introduce new terminology, I think, we should stick with JobManager (process) even though it may contain more components.

Reply:
I think, I would like to introduce new terminology ;) If I understood @StephanEwen correctly, he was also in favor of new terminology, but it is of course tricky.

Reply:
@NicoK I changed this for "JobManager Process" for now. Let's move the discussion about renaming this to the appropriate PR for the Concepts or Deployment Section.

Reply:
FWIW, I like `Flink Master Process` for the reasons outlined by Konstantin. I think it's a good call to move the discussion of whether we want to change it to another place/time.

Reply:
just FYI, I also used "Flink master process" when describing the concept in the book.